### PR TITLE
Prevent the 'u' prefix on keys

### LINKED ISSUE #218: Remove the u prefix of unicode strings on error output
I use unicode everywhere within my project, and this affects the output of Voluptuous' exceptions.

`required key not provided @ data[u'site_user_id']`

Is it possible to remove the unicode prefix?

### PR DESCRIPTION
**Pull request by jmwri**
Fixes #218

### CODE REVIEW

**Comment by tusharmakkar08** [comment]
Hey @jmwri 

Travis build is failing. Please update the PR.

Thanks

**Comment by jmwri** [comment]
@tusharmakkar08 Sorry about that, had some issues with 3.2. I've added six to install_requires.

File: voluptuous/encoding.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def to_utf8_py2(data):
+    import sys
```
Comment (tusharmakkar08) [inline voluptuous/encoding.py]: Move this import to top

File: setup.py [hunk 2]
```diff
@@ -50,5 +50,6 @@
     ],
     install_requires=[
         'setuptools >= 0.6b1',
+        'six >= 1.10'
```
Comment (tusharmakkar08) [inline setup.py]: Add it in requirements.txt

File: voluptuous/encoding.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+def to_utf8_py2(data):
```
Comment (jmwri) [inline voluptuous/encoding.py]: I'm having some trouble with this. When I move the function I'm getting lots of doctest errors.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+nose==1.3.7
```
Comment (tusharmakkar08) [inline requirements.txt]: You will have to update `setup.py` also for adding `requirements.txt` to install_requires.

File: voluptuous/encoding.py (hunk 3 above)
Comment (tusharmakkar08) [inline voluptuous/encoding.py]: What errors are you getting ? Is it due to circular dependency?

File: voluptuous/tests/tests.py [hunk 5]
```diff
@@ -8,6 +8,8 @@
     validate, ExactSequence, Equal, Unordered
 )
 from voluptuous.humanize import humanize_error
+from voluptuous.encoding import to_utf8_py2
+import six
```
Comment (tusharmakkar08) [inline voluptuous/tests/tests.py]: Can we get this done without importing `six`? Because Ideally we don't want voluptuous to have external dependencies. Check [this](https://docs.python.org/2/howto/unicode.html#the-unicode-type)

Comment (jmwri) [inline voluptuous/tests/tests.py]: I've had to make separate tests for py2 & 3 due to checking for unicode or str objects.

File: requirements.txt (hunk 4 above)
Comment (jmwri) [inline requirements.txt]: No problem

**Comment by tusharmakkar08** [comment]
@jmwri : There are few conflicts. Please resolve the conflicts and update the PR.

File: requirements.txt (hunk 4 above)
Comment (tusharmakkar08) [inline requirements.txt]: We can now remove `requirements.txt` since this is not a dependency on voluptuous but a dependency on tests.

File: setup.py [hunk 6]
```diff
@@ -22,6 +22,13 @@
 
 description = long_description.splitlines()[0].strip()
 
+try:
```
Comment (tusharmakkar08) [inline setup.py]: Please revert it back to original setup.py

File: setup.py [hunk 7]
```diff
@@ -22,7 +22,6 @@
 
 description = long_description.splitlines()[0].strip()
 
-
```
Comment (tusharmakkar08) [inline setup.py]: Please revert this change as well

File: voluptuous/util.py [hunk 8]
```diff
@@ -148,3 +150,17 @@ def __str__(self):
...
         return repr(self.lit)
+
+
+def to_utf8_py2(data):
+    if sys.version_info < (3,) and isinstance(data, unicode):
+        return data.encode('utf-8')
+    return data
+
+
+if sys.version_info < (3,):
```
Comment (tusharmakkar08) [inline voluptuous/util.py]: Why not doing something like: 

```
def u(x):
      if sys.version_info < (3, ):
           return unicode(x)
      else:
          return x
```

**Comment by tusharmakkar08** [comment]
@jmwri : Merged 🍰

**Comment by jmwri** [comment]
Yay!
